### ISSUE TITLE
New drag widget with 3 values, akin to DragFloat3 in ImGui

### ISSUE THREAD

**Issue by MindSwipe** [emilk/egui#176]
**Is your feature request related to a problem? Please describe.**
I want to list the translation of a transform in one single row and have it be editable, right now I have to add 3 separate `DragValue`s to a `horizontal` which results in non uniformly sized boxes to drag, see [this](https://imgur.com/a/ujOfxjD) screenshot for a visual aid

**Describe the solution you'd like**
A new widget called `DragValue3` which creates 3 equally sized `DragValue`s in one row. Something like [this](https://imgur.com/a/wwRvEi4)

**Describe alternatives you've considered**
Adding 3 `DragValue` s to a horizontal row. This does work, but the boxes to change the values [aren't always the same size](https://imgur.com/a/ujOfxjD)

**Comment by jakobhellermann** [comment]
I used
```rust
ui.label("Transform");
ui.columns(3, |ui| {
    ui[0].add(DragValue::f32(&mut x));
    ui[1].add(DragValue::f32(&mut y));
    ui[2].add(DragValue::f32(&mut z));
});
```
which looks like this:

![image](https://user-images.githubusercontent.com/22177966/107860541-0ffaf780-6e40-11eb-9c47-49fd38de673d.png)

**Comment by emilk** [comment]
Yeah, I agree this is something egui should have. I think the implementation should be just a horizontal layout of three DragValues, but with some fixed width of the dragvalues (would require an addiiton of `DragValue::width(...)` in the builder).

**Comment by emilk** [comment]
One problem with `DragValue3` is that people will want to use it for both x,y,z and r,g,b and will want to use it with their own `Vec3` types, so it needs a lot of customization (e.g. units) and also will be hard to make ergonomic (e.g. passing `&mut [f32; 3]` will not work for `glam::Vec3`).

**Comment by DrOptix** [comment]
For a horizontal layout is there a way to have some widgets with the minimum required width and others to extent to as much as possible?

I'm trying to shape egui widgets and layouts into somethings like this:
```
        // +------------------------------------------------------------------+
        // | Transform       X: 12345.67890   Y: 12345.67890   Z: 12345.67890 |
        // | Rotation        X: 12345.67890   Y: 12345.67890   Z: 12345.67890 |
        // | Scale           X: 12345.67890   Y: 12345.67890   Z: 12345.67890 |
        // +------------------------------------------------------------------+
```

Something like Unity or Unreal transform component editor.

**Comment by emilk** [comment]
> For a horizontal layout is there a way to have some widgets with the minimum required width and others to extent to as much as possible?

A horizontal layout has no memory, and so only the last element can be set to "use as much as is available" (the earlier widgets can't know how much later widgets will use).

`Grid` could theoretically support this (since it remembers the sizes of the columns from previous frame), but currently it does not.
